[messagingframework] Use upstream libaccounts manager.

dcaliste opened the pull request:
This reworks the patchs to use the new upstream libaccounts plugin.

It can be used with https://codereview.qt-project.org/c/qt-labs/messagingframework/+/630364 for testing.

If upstream gets accepoted, I'll rework the remaining parts of 0002 patch for them to be consistent patchs, potentially upstreamable.

pvuorela commented:
In this form the accounts-qt integration patch commit message could be adjusted to match the updated content. Now referring to lots of things which it no longer has.

Nicely small delta to upstream with all this!

pvuorela commented:
Doesn't compile for me: 
Cannot find file: /home/pvuorela/src/messagingframework/upstream/src/plugins/mailaccountmanagers/accounts/accounts.pro.
make[2]: *** [Makefile:225: sub-mailaccountmanagers-accounts-make_first] Error 2

dcaliste commented:
Oops, sorry, the directory was first called `sso`, and I forgot to rename the `pro` file. Funnily enough, just typing `make` in the build dir never noticed the name change when I renamed the directory to `accounts`. Now, following your review advice, I renamed it once again to `libaccounts` and I also renamed the `pro` file this time.

pvuorela commented:
To be clear: is the draft here as in waiting for upstream to be merged or something more?

dcaliste commented:
Hello, yes, the draft is for "waiting for upstream convergence on API for account plugin and subsequent changes", not "I'm going to add further changes". Maybe it's a misuse on Draft: mention from my side. I can remove it if you think it's misleading.

pvuorela commented:
No need to adjust now that the interpretation is clear.

pvuorela commented on rpm/0002-Accounts-qt-integration.patch:
```diff
@@ -84,1966 +84,74 @@ This mirrors the behavour of qmf-oauth2-plugin. See
-     }
-+#endif
+-            LessThanFunctorA lessThan(d->sortKey);
++            if (!d->idList.contains(id)) {
```
No need to adjust now, but just noting that even the remaining delta could be smaller by making the contains() checks/filtering first and then just keeping the upstream handler. Or wondering if even upstream should be having some of that. Enough changes even without that but just saying :)

dcaliste replied:
Indeed, I'll treat the remaining chunks in this patch in a seoparate PR, just focussed on what is left. I've adjusted the description of the patch accordingly.

pvuorela commented on rpm/0003-Start-messageserver-on-system-startup-in-case-there-.patch:
```diff
@@ -161,13 +161,13 @@ index 00000000..0d583d15
  SUBDIRS = messageserver
 +
-+contains(DEFINES,USE_ACCOUNTS_QT) {
++packagesExist(accounts-qt5) {
```
Hm, also upstream compiles this if found, but that's maybe slightly dangerous as if affects the resulting libraries that much. Though maybe accounts-qt is not found much by accident.

Anyhow, the accounts integration patch does still have one #ifdef USE_ACCOUNTS_QT in servicehandler.cpp and the .spec is building with it. If it's getting automatically used, it should then happen also there?

pvuorela replied:
No biggie but guess the explicit opt-in could be nicer, especially since the build process doesn't that well tell what options it's using. Cmake qt6 notes it didn't find accounts-qt6 but that's about it.

Either way good if the acconts-qt usage is consistent. Either fully automatic or then always opt-in.

dcaliste replied:
I've reverted this, so it's still using the USE_ACCOUNTS_QT flag.

pvuorela commented on rpm/0008-Retrieve-message-lists-based-on-the-folder-sync-poli.patch:
```diff
@@ -8,16 +8,88 @@ when downloading messages from different folders.
+     return false;
+ }
+ 
++QMailAccount::FolderSyncPolicy folderSyncPolicyToEnum(QString const &policy)
```
Seems just moving code around, but more commonly these would be const QString &. No need to change, but wouldn't mind if done.

dcaliste replied:
I think the code move comes from the regeneration of the patch with git format-patch now that the patch is applied to the file libaccount_p.cpp and not qmailstore_p.cpp anymore.

I've change the declaration to be const QString &.

pvuorela reviewed (commented):
Seems fine here, just some small comments. And then the earlier one about accounts integration patch commit message still applies.

pvuorela commented on rpm/qmf-qt5.spec:
```diff
@@ -173,6 +173,7 @@ touch .git
     DEFINES+=QMF_NO_MESSAGE_SERVICE_EDITOR \
     DEFINES+=QMF_NO_WIDGETS \
     DEFINES+=USE_ACCOUNTS_QT \
+    DEFINES+=QMF_ACCOUNT_MANAGER_CLASS=SSOAccountManager \
```
This part doesn't compile now. Adjusting the name gets better success but would be if build wouldn't need to know class names. There's at least now the USE_ACCOUNTS_QT passed even if the option is halfway automatic and halfway using that.

dcaliste replied:
I've removed this finally and use only USE_ACCOUNTS_QT flag. Then, internally, it is setting the QMF_ACCOUNT_MANAGER_CLASS to LibAccountManager. So now it's again fully opt-in.

dcaliste commented:
I still need to update the upstream submodule. I will do when it's merged in upstream.

pvuorela commented:
Seems good! I'll do the final approve and merge once the submodule checkout is the final one.

dcaliste commented:
Thank you. I've updated the submodule pointer.

pvuorela reviewed (approved):
Yay, finally getting there \o/
